auth.py

```diff
@@ -33,6 +34,22 @@ def metadata():
     return resp
 
 
+def check_or_create_user(user):
+    db_session = Session()
+    try:
+        is_user = db_session.query(User).filter(User.email == user.email).first()
+        if is_user is None:
+            user = User(name=user.name, first_name=user.first_name, email=user.email)
+            db_session.add(user)
+            db_session.commit()
+            is_user = db_session.query(User).filter(User.email == user.email).first()
```

Comment:
You may be able to avoid duplicating the request by doing `db_session.refresh(user)`.